pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -730,10 +732,13 @@ private void handleDedup(ImmutableSegmentImpl immutableSegment) {
     }
   }
 
-  private void handleUpsert(ImmutableSegment immutableSegment) {
+  private void handleUpsert(ImmutableSegment immutableSegment, @Nullable SegmentZKMetadata zkMetadata) {
     String segmentName = immutableSegment.getSegmentName();
     _logger.info("Adding immutable segment: {} with upsert enabled", segmentName);
 
+    // Set the ZK creation time for upsert consistency across replicas
```

Comment:
nit: make the comment a bit more specific: "Set the ZK creation time so that same creation time can be used to break the comparison ties across replicas, to ensure data consistency of replicas"